nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/TestAttributesToCSV.java

```diff
@@ -0,0 +1,540 @@
+        assertTrue(csvAttributeValues.contains("\"This is our family's favorite beach. We highly recommend it. \n\nThanks, Jim\""));
+
+        assertTrue(csvAttributeValues.contains(flowFile.getAttribute("filename")));
+        assertTrue(csvAttributeValues.contains(flowFile.getAttribute("path")));
+        assertTrue(csvAttributeValues.contains(flowFile.getAttribute("uuid")));
+    }
+
+    @Test
+    public void testNoAttrListNoCoreNullOffToContent() throws IOException {
+        final TestRunner testRunner = TestRunners.newTestRunner(new AttributesToCSV());
+        testRunner.setProperty(AttributesToCSV.DESTINATION, OUTPUT_OVERWRITE_CONTENT);
+        testRunner.setProperty(AttributesToCSV.INCLUDE_CORE_ATTRIBUTES, "false");
+        testRunner.setProperty(AttributesToCSV.NULL_VALUE_FOR_EMPTY_STRING, "false");
+
+        ProcessSession session = testRunner.getProcessSessionFactory().createSession();
+        FlowFile ff = session.create();
+
+        ff = session.putAttribute(ff, "beach-name", "Malibu Beach");
+        ff = session.putAttribute(ff, "beach-location", "California, US");
+        ff = session.putAttribute(ff, "beach-endorsement", "This is our family's favorite beach. We highly recommend it. \n\nThanks, Jim");
+
+        testRunner.enqueue(ff);
+        testRunner.run();
+
+        List<MockFlowFile> flowFilesForRelationship = testRunner.getFlowFilesForRelationship(AttributesToCSV.REL_SUCCESS);
+
+        testRunner.assertTransferCount(AttributesToCSV.REL_FAILURE, 0);
+        testRunner.assertTransferCount(AttributesToCSV.REL_SUCCESS, 1);
+
+        MockFlowFile flowFile = flowFilesForRelationship.get(0);
+
+        assertEquals(OUTPUT_MIME_TYPE, flowFile.getAttribute(CoreAttributes.MIME_TYPE.key()));
+
+        final byte[] contentData = testRunner.getContentAsByteArray(flowFile);
+
+        final String contentDataString = new String(contentData, "UTF-8");
+        Set<String> contentValues = new HashSet<>(getStrings(contentDataString));
+
+        assertEquals(3, contentValues.size());
+
+        assertTrue(contentValues.contains("Malibu Beach"));
+        assertTrue(contentValues.contains("\"California, US\""));
+        assertTrue(contentValues.contains("\"This is our family's favorite beach. We highly recommend it. \n\nThanks, Jim\""));
+
+    }
+
+
+    @Test
+    public void testAttrListNoCoreNullOffToAttribute() throws IOException {
+        final TestRunner testRunner = TestRunners.newTestRunner(new AttributesToCSV());
+        testRunner.setProperty(AttributesToCSV.DESTINATION, OUTPUT_NEW_ATTRIBUTE);
+        testRunner.setProperty(AttributesToCSV.INCLUDE_CORE_ATTRIBUTES, "false");
+        testRunner.setProperty(AttributesToCSV.ATTRIBUTES_LIST, "beach-name,beach-location,beach-endorsement");
+        testRunner.setProperty(AttributesToCSV.NULL_VALUE_FOR_EMPTY_STRING, "false");
+
+        ProcessSession session = testRunner.getProcessSessionFactory().createSession();
+        FlowFile ff = session.create();
```

Comment:
See other comment about this. You can wipe out these references and use the cleaner map-driven approach.